### PR TITLE
Added pydocstyle and docformatter in pre-commit config

### PR DESCRIPTION
**Pull request by karan6181**
## Description
- Added pydocstyle and docformatter in pre-commit config
- Added docstring as part of it
- Update the existing docstrings and comments.

### CODE REVIEW

File: docs/source/conf.py [hunk 1]
```diff
@@ -371,13 +371,11 @@ def _auto_rst_for_module(module: types.ModuleType, exclude_members: List[Any]) -
...
 def _modules_to_rst() -> List[types.ModuleType]:
-    """Return the list of modules for which to generate API reference rst
-    files."""
-
+    """Return the list of modules for which to generate API reference rst files."""
     document_modules: List[types.Module] = [
         streaming,
     ]
-    exclude_modules: List[types.Module] = []
+    exclude_modules: List[types.Module] = [streaming.base, streaming._version]
```
Comment (knighton) [inline docs/source/conf.py:378]: why? there is a lot of documented code under base which does not happen to be imported to top-level

Comment (knighton) [reply docs/source/conf.py:378]: also the docs are missing for streaming.text, streaming.vision, we need to fix this too (__all__?)

Comment (karan6181) [reply docs/source/conf.py]: Added text and vision to a doc in https://github.com/mosaicml/streaming/pull/20/commits/b9e4a2d410dea27e5b08949e7ddca132ee29444d. Looks like adding all the class and method as part of `base` require some thorough investigation. I will do that in a separate PR.

File: streaming/__init__.py [hunk 2]
```diff
@@ -1,10 +1,15 @@
...
+import streaming.text as text
+import streaming.vision as vision
 from streaming._version import __version__
 from streaming.base import (CSVWriter, Dataset, JSONWriter, LocalDataset, MDSWriter, TSVWriter,
                             XSVWriter)
 
 __all__ = [
-    'Dataset', 'CSVWriter', 'JSONWriter', 'MDSWriter', 'TSVWriter', 'XSVWriter', 'LocalDataset'
+    'Dataset', 'CSVWriter', 'JSONWriter', 'MDSWriter', 'TSVWriter', 'XSVWriter', 'LocalDataset',
+    'vision', 'text'
```
Comment (knighton) [inline streaming/__init__.py:14]: i assume we want `base` too here

Comment (karan6181) [reply streaming/__init__.py:14]: I tried that too but the docs is not generating for sub-directory. I will take a look in a separate PR to make it to work.

**Review by knighton — APPROVED** [review]

**Pull request merged**